### PR TITLE
Fix redirection of stdout and stderr (bnc#943757)

### PR DESCRIPTION
**Pull request by ancorgs**
Just a possible alternative. See #41 for the other one.

Checked that it fixes the problem by means of 
https://github.com/ancorgs/yast-yast2/commit/f52bc2350e9690415d1f53b5175e107a7529433d

### CODE REVIEW

**Comment by shundhammer** [comment]
LGTM

**Comment by mvidner** [comment]
We are saving and passing around *pointers* to streambufs, but in http://www.cplusplus.com/reference/ios/ios/rdbuf/ I couldn't quite find a guarantee that the pointer will remain valid.

**Comment by ancorgs** [comment]
@mvidner Suggestions to extend the test? Right now we check stderr is redirected, what should I add to the test in order to test that it's restored when needed?

**Comment by mvidner** [comment]
I don't know yet, but this article seems to address that: http://wordaligned.org/articles/cpp-streambufs

**Comment by mvidner** [comment]
Apparently the streambuf is valid as long as the strem is valid. So that is why this example from the above article will segfault: `log` is destroyed before the global `cout` and then `cout` tries to flush its buffer.
```c++
#include <fstream>
#include <iostream>

int main()
{
    std::ofstream log("oops.log");
    std::cout.rdbuf(log.rdbuf());
    std::cout << "Oops!\n"; // SEGV even without this line
    return 0;
}
```

File: src/NCurses.cc [hunk 1]
```diff
@@ -555,24 +555,22 @@ void NCurses::ForgetDlg( NCDialog * dlg_r )
...
 
     yuiMilestone() << "isatty(stderr)" << ( isatty( 2 ) ? "yes" : "no" ) << std::endl;
 
     if ( isatty( 2 ) && theTerm )
     {
-	// redirect stderr to log
-	close( 2 );
-	open( log.c_str(), O_APPEND | O_CREAT, 0666 );
+	// redirect stderr to /dev/null
+	std::cerr.rdbuf(devNull.rdbuf());
```
Comment (mvidner) [inline src/NCurses.cc]: This is not supposed to work, as `cerr` will we using the streambuf of devNull that will be destroyed when it goes out of scope. In practice it seems to work but anyway I would prefer the POSIX approach of #41.

**Comment by ancorgs** [comment]
Closed in favor of #41

**Pull request closed without merging**